tools/python/boutiques/localExec.py

```diff
@@ -902,9 +903,12 @@ def escape_string(s):
                             s_val += s
                         else:
                             s_val += s + list_sep
-                    val = s_val
-                elif escape:
-                    val = escape_string(val)
+                    val = s_val if surr_char is None else s_val + surr_char
+                else:
+                    if escape:
+                        val = escape_string(val)
+                    if surr_char is not None:
+                        val = surr_char + str(in_out_dict[param_id]) + surr_char
```

Comment:
This breaks the escaping of special characters in case `surr_char` isn't the double quote character. 
`val` should always be escaped after `surr_char` was appended. This is also a problem with lists. For instance, the command line generated from https://gist.github.com/glatard/a9d1fafb35d21a4fcbf111eedd5c0b60 should be `echo '-file name with space.tex-' '-filename with a ;.pdf-'`, and not `echo -file name with space.tex- -filename with a ;.pdf-`

Reply:
Ok, so would it be sufficient to just add `val = escape_string(val)` after adding the surrounding characters?

Reply:
Yes I think so, but only `if escape`. For instance, numbers shouldn't be escaped.